File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/ServerTest.cs
namespace Nancy.ViewEngines.React.UnitTests.ConfigurationTests
{
    using Xunit;

    public class ServerTest : TestBase
    {
        public ServerTest()
            : base("Server.config")
        {
        }

        [Fact]
        public void Server_configuration_should_load_properties()
        {
            Assert.Equal("assets-path", this.Configuration.Server.Assets.Path);
        }
    }
}

File: Nancy.ViewEngines.React.UnitTests/ConfigurationTests/ServerTest.js
import path from 'path';
import expect from 'expect.js';
import { readFile } from '../../Nancy.ViewEngines.React/tools/commands/file';
import { parseConfig } from '../../Nancy.ViewEngines.React/tools/commands/options';

describe('Server test', () => {
  let parse;

  beforeEach(() => {
    const file = path.resolve(__dirname, '../ConfigurationFixtures', 'Server.config');
    parse = readFile(file).then(parseConfig);
  });

  it('should load properties', () =>
    parse.then(config => {
      expect(config.server.assets.path).to.be('assets-path');
    }));
});
